reana_job_controller/kubernetes_job_manager.py

```diff
@@ -79,7 +79,6 @@ def execute(self):
                 'namespace': K8S_DEFAULT_NAMESPACE
             },
             'spec': {
-                'backoffLimit': current_app.config['MAX_JOB_RESTARTS'],
```

Comment:
@tiborsimko failed jobs should not restart anymore.

Reply:
How is removing it going to help if the [default value is 6](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#jobspec-v1-batch)? I randomly bumped into a project which deploys CWL inside Kubernetes and what they do there is to set `backoffLimit` to 0 because they do not want to retry.

I guess that what we want to do is to retry, but from the same pod, not to create one on each retry. We didn't do this is the past because `onFailure` didn't work together with `backoffLimit`, see commit content and message https://github.com/reanahub/reana-job-controller/commit/dce8aa1462e6f519671b661e89a8798b3b015b06. So if what we want to retry on the same pod we should use `onFailure`.

Reply:
Thanks for the suggestion! Indeed it helped.


```
 $ kubectl describe pod 9cf97d5d-92af-4420-91e0-fa378b1e78a1-plw77
...
Events:
  Type     Reason     Age               From               Message
  ----     ------     ----              ----               -------
  Normal   Scheduled  62s               default-scheduler  Successfully assigned default/9cf97d5d-92af-4420-91e0-fa378b1e78a1-plw77 to minikube
  Normal   Pulling    60s               kubelet, minikube  Pulling image "python:2.7-slim"
  Normal   Pulled     51s               kubelet, minikube  Successfully pulled image "python:2.7-slim"
  Normal   Created    6s (x4 over 51s)  kubelet, minikube  Created container 9cf97d5d-92af-4420-91e0-fa378b1e78a1
  Normal   Started    6s (x4 over 50s)  kubelet, minikube  Started container 9cf97d5d-92af-4420-91e0-fa378b1e78a1
  Normal   Pulled     6s (x3 over 49s)  kubelet, minikube  Container image "python:2.7-slim" already present on machine
  Warning  BackOff    5s (x5 over 48s)  kubelet, minikube  Back-off restarting failed container

$ kubectl get pods
NAME                                                      READY   STATUS        RESTARTS   AGE
9cf97d5d-92af-4420-91e0-fa378b1e78a1-plw77                0/1     Terminating   3          63s
```

Reply:
The policy is that (1) we don't want to silently retry any lengthy user scientific calculations (this is quite like `backoffLimit 0`) except for retrying some internal platform things such as (2) submitting a job to HTCondor `schedd` if the first submission failed (in which case we may want to retry a bit later). Is it easy to distinguish the two? If not, we may prefer to always fail without retrying...